### PR TITLE
feat(flux): add OCIRepository support

### PR DESCRIPTION
**Pull request by onedr0p**
Signed-off-by: Devin Buhl <devin@buhl.casa>

## Changes

Add OCIRepository support - see https://fluxcd.io/flux/components/source/ocirepositories/

## Context

- #18509

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by onedr0p** [comment]
First time contributor, please be kind <3 Hopefully this makes sense.

File: lib/modules/manager/flux/__fixtures__/oci-repository.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: source.toolkit.fluxcd.io/v1beta2
```
Comment (viceice) [inline lib/modules/manager/flux/__fixtures__/oci-repository.yaml:1]: please inline small fixtures

File: lib/modules/manager/flux/extract.ts [hunk 2]
```diff
@@ -32,8 +33,9 @@ function readManifest(content: string, file: string): FluxManifest | null {
   const manifest: FluxManifest = {
     kind: 'resource',
     file,
-    releases: [],
-    repositories: [],
+    helmReleases: [],
```
Comment (viceice) [inline lib/modules/manager/flux/extract.ts:36]: please don't rename if not required, or do separate refactor PR.

File: lib/modules/manager/flux/index.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
+import { DockerDatasource } from '../../datasource/docker';
 import { GithubReleasesDatasource } from '../../datasource/github-releases';
 import { HelmDatasource } from '../../datasource/helm';
 import { systemManifestRegex } from './common';
 
-export { extractAllPackageFiles, extractPackageFile } from './extract';
 export { updateArtifacts } from './artifacts';
+export { extractAllPackageFiles, extractPackageFile } from './extract';
```
Comment (viceice) [inline lib/modules/manager/flux/index.ts:7]: why this change?

File: lib/modules/manager/flux/types.ts [hunk 4]
```diff
@@ -37,16 +37,31 @@ export interface HelmRepository extends KubernetesResource {
   };
 }
 
-export type FluxResource = HelmRelease | HelmRepository;
+export interface OCIRepository extends KubernetesResource {
```
Comment (viceice) [inline lib/modules/manager/flux/types.ts:40]:
```suggestion
export interface OciRepository extends KubernetesResource {
```
would prefer to not have too many upper case letters.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/flux/index.ts (hunk 3 above)
Comment (onedr0p) [reply lib/modules/manager/flux/index.ts:7]:
```
npm test
...
...
/workspaces/renovate/lib/modules/manager/flux/extract.spec.ts
  2:1  error  `.` import should occur after import of `../types`  import/order

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

File: lib/modules/manager/flux/extract.ts (hunk 2 above)
Comment (onedr0p) [reply lib/modules/manager/flux/extract.ts:36]: Any chance for an exception here? repositories can be many things in Flux.

File: lib/modules/manager/flux/types.ts (hunk 4 above)
Comment (onedr0p) [reply lib/modules/manager/flux/types.ts:40]: You are okay with having the TS interface name not match the Flux `OCIRepository` resource?

File: lib/modules/manager/flux/index.ts (hunk 3 above)
Comment (onedr0p) [reply lib/modules/manager/flux/index.ts:7]: VSCode on save moves it back to the position it was but than that error happens on npm test, is there a way around that?

File: lib/modules/manager/flux/extract.ts (hunk 2 above)
Comment (onedr0p) [reply lib/modules/manager/flux/extract.ts:36]: I will open a refactor PR.

File: lib/modules/manager/flux/types.ts (hunk 4 above)
Comment (onedr0p) [reply lib/modules/manager/flux/types.ts]: Done.

File: lib/modules/manager/flux/index.ts (hunk 3 above)
Comment (onedr0p) [reply lib/modules/manager/flux/index.ts:7]: Fixed.

**Comment by onedr0p** [comment]
Closing this to submit a refactor PR in first.
